### PR TITLE
#3804 create new logic to get all locations

### PR DESCRIPTION
**Pull request by Bohdan-Melnyk**
Add neAPI for pricing page

### CODE REVIEW

File: core/src/main/java/greencity/controller/SuperAdminController.java [hunk 1]
```diff
@@ -446,4 +446,21 @@ public SuperAdminController(SuperAdminService superAdminService) {
...
+     * @return {@link GetTariffsInfoDto}
+     * @author Bohdan Melnyk
+     */
+    @ApiOperation(value = "Get all info about tariffs.")
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = HttpStatuses.OK),
+        @ApiResponse(code = 401, message = HttpStatuses.UNAUTHORIZED),
+        @ApiResponse(code = 403, message = HttpStatuses.FORBIDDEN)
+    })
+    @GetMapping("/getAllTariffs")
```
Comment (VadymMakitra) [inline core/src/main/java/greencity/controller/SuperAdminController.java]: It's not rest style please check https://restfulapi.net/

File: dao/src/main/java/greencity/entity/order/Bag.java [hunk 2]
```diff
@@ -57,4 +57,7 @@
 
     @OneToMany(mappedBy = "bag", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
     private List<BagTranslation> bagTranslations;
+
+    @ManyToOne
+    TariffsInfo tariffsInfo;
```
Comment (VadymMakitra) [inline dao/src/main/java/greencity/entity/order/Bag.java]: You were forgetting about access modifier;

File: dao/src/main/java/greencity/entity/order/CourierLocation.java [hunk 3]
```diff
@@ -45,4 +46,7 @@
 
     @OneToMany(fetch = FetchType.LAZY, cascade = CascadeType.ALL, mappedBy = "courierLocations")
     List<Order> orders;
+
+    @ManyToOne
+    TariffsInfo tariffsInfo;
```
Comment (VadymMakitra) [inline dao/src/main/java/greencity/entity/order/CourierLocation.java]: You were forgetting about access modifier;

File: dao/src/main/java/greencity/entity/order/Service.java [hunk 4]
```diff
@@ -52,4 +52,7 @@
 
     @OneToMany(mappedBy = "service", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
     List<ServiceTranslation> serviceTranslations;
+
+    @ManyToOne
+    private TariffsInfo tariffsInfo;
```
Comment (VadymMakitra) [inline dao/src/main/java/greencity/entity/order/Service.java:57]: !

File: dao/src/main/resources/db/changelog/logs/20220902-ch-add-table-tariffs-info-Melnyk.xml [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+                <constraints nullable="false"/>
+            </column>
+
+        </createTable>
+
+<!--        <addForeignKeyConstraint baseColumnNames="tariffs_info_id"-->
+<!--                                 baseTableName="courier_locations"-->
+<!--                                 referencedColumnNames="id"-->
+<!--                                 referencedTableName="tariffs_info"-->
+<!--                                 constraintName="blabla"/>-->
```
Comment (VadymMakitra) [inline dao/src/main/resources/db/changelog/logs/20220902-ch-add-table-tariffs-info-Melnyk.xml]: Remove comment's;

File: service/src/main/java/greencity/mapping/GetTariffsInfoDtoMapper.java [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Component
+public class GetTariffsInfoDtoMapper extends AbstractConverter<TariffsInfo, GetTariffsInfoDto> {
+    @Override
+    protected GetTariffsInfoDto convert(TariffsInfo source) {
+        List<CourierLocation> courierLocations = source.getCourierLocations();
+        List<Location> locations = courierLocations.stream()
```
Comment (VadymMakitra) [inline service/src/main/java/greencity/mapping/GetTariffsInfoDtoMapper.java:29]: Will be better if you write here source source.getCourierLocations().stream() and remove courierLocations;

**Review by VadymMakitra — CHANGES_REQUESTED** [review]

**Comment by VadymMakitra** [comment]
Resolve comments

File: service/src/main/java/greencity/service/ubs/SuperAdminServiceImpl.java [hunk 7]
```diff
@@ -453,6 +453,14 @@ public void addLocationToCourier(NewLocationForCourierDto dto) {
         courierLocationRepository.save(courierLocation);
     }
 
+    @Override
+    public List<GetTariffsInfoDto> getAllTariffsInfo() {
+        return tariffsInfoRepository.findAll()
+            .stream()
+            .map(i -> modelMapper.map(i, GetTariffsInfoDto.class))
```
Comment (VadymMakitra) [inline service/src/main/java/greencity/service/ubs/SuperAdminServiceImpl.java]: rename variable i;

**Review by VadymMakitra — CHANGES_REQUESTED** [review]

**Review by Max877866 — APPROVED** [review]

**Review by Natalia1910 — APPROVED** [review]

File: dao/src/main/java/greencity/entity/enums/LocationStatus.java [hunk 8]
```diff
@@ -1,5 +1,23 @@
 package greencity.entity.enums;
 
 public enum LocationStatus {
-    ACTIVE, DEACTIVATED
+    ACTIVE("Активно"), DEACTIVATED("Неактивно");
+
+    private String status;
+
+    LocationStatus() {
```
Comment (thesvider) [inline dao/src/main/java/greencity/entity/enums/LocationStatus.java]: you can use lombok annotations for constructors and methods

**Review by KuzbytMaksym — APPROVED** [review]

**Review by Natalia1910 — APPROVED** [review]

**Review by Max877866 — APPROVED** [review]

**Review by VadymMakitra — APPROVED** [review]

File: dao/src/main/java/greencity/entity/user/User.java [hunk 9]
```diff
@@ -67,4 +68,7 @@
 
     @Column(name = "date_of_registration")
     private LocalDate dateOfRegistration;
+
+    @OneToMany(fetch = FetchType.LAZY, mappedBy = "creator")
```
Comment (thesvider) [inline dao/src/main/java/greencity/entity/user/User.java:72]: I think fetch type is Lazy by default, no need to declare

File: dao/src/main/resources/db/changelog/logs/20221002-ch-add-column-creator-id-in-tariffs-info-table-Melnyk.xml [hunk 10]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+                   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                   xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.4.xsd">
+    <changeSet id="Melnyk-7" author="Bohdan Melnyk">
+
+        <addColumn tableName="tariffs_info">
```
Comment (thesvider) [inline dao/src/main/resources/db/changelog/logs/20221002-ch-add-column-creator-id-in-tariffs-info-table-Melnyk.xml:7]: why are you keeping separately add column for tariffs_info? You've created this table in 20220902-ch-add-table-tariffs-info-Melnyk.xml so why don't you declare all the needed columns there?

File: service/src/main/java/greencity/mapping/GetTariffsInfoDtoMapper.java [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+            .map(CourierLocation::getLocation)
+            .collect(Collectors.toList());
```
Comment (thesvider) [inline service/src/main/java/greencity/mapping/GetTariffsInfoDtoMapper.java:31]: can you simplify this method? create a few private methods instead.

**Pull request merged**